### DISCUSSION TITLE
Stop optimize if trials are not improving

### DISCUSSION THREAD

**Discussion by Ohadkaiser** [optuna/optuna#4296] [Q&A]
### Don't use GitHub Issues to ask support questions.

Hi, I'm no sure if there is already an issue for that, but I'm trying to stop optimize if all my trails aren't improving the score of the best study yet. for example, stopping after 10 trails in a row that their value isn't the best value

**Comment by Ohadkaiser** [comment]
I don't have an intermediate result, thus I can't use pruning

**Reply by nzw0301** [reply]
I suppose you don't call `trial.report` if your trials don't have intermediate values. Please see https://optuna.readthedocs.io/en/stable/reference/generated/optuna.trial.Trial.html#optuna.trial.Trial.report.

**Reply by Ohadkaiser** [reply]
How can I use report if I don't have anyway to prune the trial?

**Reply by nzw0301** [reply]
For example, call `raise optuna.TrialPruned()` as in the example.

```python
def objective(trial):
    ....
    # call suggest methods and compute `objective_value`
    if SOME_PRUNE_CONDITION:
        raise optuna.TrialPruned()
    return objective_value
```

**Reply by Ohadkaiser** [reply]
But I would like to get the best value for the current study, but this is not an option because you don't have the study variable in the objective

**Reply by nzw0301** [reply]
Ah, sorry I misunderstood your question. i believe that the topic is discussed in https://github.com/optuna/optuna/issues/1001.

**Reply by Ohadkaiser** [reply]
Thanks! I will use it 😊
